hepcrawl/parsers/jats.py

```diff
@@ -507,53 +516,98 @@ def get_reference(self, ref_node):
+            ]
+
+            for xpath, field_handler in fields:
+                value = citation_node.xpath(xpath).extract_first()
+                citation_node.xpath(xpath)
+                if value:
+                    field_handler(value)
+
+            remainder = remove_tags(
+                    citation_node,
+                    strip='self::person-group'
+                          '|self::pub-id'
+                          '|self::article-title'
+                          '|self::volume'
+                          '|self::issue'
+                          '|self::year'
+                          '|self::label'
+                          '|self::publisher-name'
+                          '|self::source[../@publication-type!="proc"]'
+                          '|self::object-id'
+                          '|self::page-range'
+                          '|self::issn'
+                ).strip('"\';,. \t\n\r').replace('()', '')
+            if remainder:
+                builder.add_misc(remainder)
+
+            for editor in self.get_reference_authors(citation_node, 'editor'):
+                builder.add_author(editor, 'editor')
+
+            for author in self.get_reference_authors(citation_node, 'author'):
+                builder.add_author(author, 'author')
+
+            page_range = citation_node.xpath('./page-range/text()').extract_first()
+            if page_range:
+                page_artid = split_page_artid(page_range)
+                builder.set_page_artid(*page_artid)
+
+            yield builder.obj
+
+
+    def attach_fulltext_document(self, file_name, url):
+        self.builder.add_document(file_name, url, fulltext=True, hidden=True)
 
-        for xpath, field_handler in fields:
-            value = citation_node.xpath(xpath).extract_first()
-            if value:
-                field_handler(value)
 
-        for editor in self.get_reference_authors(citation_node, 'editor'):
-            builder.add_author(editor, 'editor')
+    @staticmethod
+    def _sanitize_collaboration_name(collaboration):
+        """Sanitize collaboration name to conform with the schema.
 
-        for author in self.get_reference_authors(citation_node, 'author'):
-            builder.add_author(author, 'author')
+        Collaboration name should have the Collaboration suffix dropped,
+        so "ATLAS", and not "ATLAS Collaboration"
 
-        page_range = citation_node.xpath('./page-range/text()').extract_first()
-        if page_range:
-            page_artid = split_page_artid(page_range)
-            builder.set_page_artid(*page_artid)
+        Args:
+            collaboration (string): collaboration name
 
-        return builder.obj
+        Returns:
+            string: sanitized collaboration name
+        """
+        if collaboration.startswith('(') and collaboration.endswith(')'):
+            collaboration = collaboration[1:-1]
+        collaboration = re.sub(r'\sCollaboration', '', collaboration)
```

Comment:
this is already handled by the builder. The parentheses handling seems more generally useful, and should be handled there too.